File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/operations/ModOperation.java
package org.evomaster.client.java.controller.mongo.operations;

/**
 * Represent $mod operation.
 * Select documents where the value of a field divided by a divisor has the specified remainder.
 */
public class ModOperation extends QueryOperationWithField {
    private final Long divisor;
    private final Long remainder;

    public ModOperation(String fieldName, Long divisor, Long remainder) {
        super(fieldName);
        this.divisor = divisor;
        this.remainder = remainder;
    }

    public Long getDivisor() {
        return divisor;
    }

    public Long getRemainder() {
        return remainder;
    }
}

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/operations/ModOperation.kt
package org.evomaster.client.java.controller.mongo.operations

/**
 * Represent $mod operation.
 * Select documents where the value of a field divided by a divisor has the specified remainder.
 */
open class ModOperation(
    open val fieldName: String,
    open val divisor: Long,
    open val remainder: Long
) : QueryOperation()